### PR TITLE
[QUIC] Latest msquic interop

### PR DESCRIPTION
**Pull request by ManickaP**
https://github.com/microsoft/msquic/pull/2741 follow up.
We can also bump dotnet/msquic to be on the same version, done: https://github.com/dotnet/msquic/pull/83

### CODE REVIEW

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/msquic_generated.cs [hunk 1]
```diff
@@ -7,6 +7,13 @@
 
 #pragma warning disable CS0649
 
+// Polyfill for MemoryMarshall on .NET Standard
```
Comment (jkotas) [inline src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/msquic_generated.cs:10]: Typo: MemoryMarshall

Comment (ManickaP) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/msquic_generated.cs:10]: cc @ThadHouse

Comment (thhous-msft) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/msquic_generated.cs:10]: Whoops. I'll fix on our end.

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/msquic_generated_macos.cs [hunk 2]
```diff
@@ -5,6 +5,8 @@
 //
 #pragma warning restore IDE0073
 
+using System.Runtime.InteropServices;
```
Comment (thhous-msft) [inline src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/msquic_generated_macos.cs:8]: What is this using added for? Its not on our side. I can add it if necessary, but I'd like to keep these as identical as possible.

Comment (ManickaP) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/msquic_generated_macos.cs:8]: https://github.com/microsoft/msquic/blob/8cba1150674cbecda7a136f344acc64c65f49932/src/cs/lib/msquic_generated_macos.cs#L8

It's on your side. Or am I going crazy? :smile:

Comment (thhous-msft) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/msquic_generated_macos.cs:8]: Oh I was looking at the windows file... So I'm going crazy. Let me remove it, theres no reason for it to be there.

**Comment by thhous-msft** [comment]
https://github.com/microsoft/msquic/pull/2759

Has a fix for both comments

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/msquic_generated.cs (hunk 1 above)
Comment (ManickaP) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/msquic_generated.cs]: Updated.

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/msquic_generated_macos.cs (hunk 2 above)
Comment (ManickaP) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/msquic_generated_macos.cs]: Updated.

**Review by CarnaViire — APPROVED** [review]
LGTM

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/msquic_extensions.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+//
+#pragma warning restore IDE0073
+
+using System;
+
+namespace Microsoft.Quic
+{
+    internal unsafe static class MsQuicExtensions
```
Comment (ManickaP) [inline src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/msquic_extensions.cs:12]:
```suggestion
    internal static unsafe class MsQuicExtensions
```

**Comment by ThadHouse** [comment]
You should exclude the polyfill file. It should never be needed, and we don’t want it accidentally being used.

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/msquic_extensions.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+//
+#pragma warning restore IDE0073
+
+using System;
+
+namespace Microsoft.Quic
+{
+    internal static unsafe class MsQuicExtensions
+    {
+        public static void SetConnectionCallback(this ref QUIC_API_TABLE Table, QUIC_HANDLE* Handle, delegate* unmanaged[Cdecl]<QUIC_HANDLE*, void*, QUIC_CONNECTION_EVENT*, int> Callback, void* Context)
```
Comment (rzikm) [inline src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/msquic_extensions.cs:14]: Should we also update our code to use these, or can it wait to be opportunistically included in some other PR?

**Review by rzikm — APPROVED** [review]
LGTM

**Pull request merged**